functions/index.js

```diff
@@ -1,8 +1,361 @@
 const functions = require('firebase-functions');
+const fs = require('fs');
+const path = require('path');
+var https = require("https");
+var url = require('url');
 
-// Create and Deploy Your First Cloud Functions
-// https://firebase.google.com/docs/functions/write-firebase-functions
+// travelData is the sample data that we use to demo filtering on the frontend.
+const src = path.join(path.dirname(__filename), 'travel-data.json');
+const travelData = JSON.parse(fs.readFileSync(src, 'utf8'));
 
-exports.helloWorld = functions.https.onRequest((request, response) => {
- response.send("Hello from Firebase!");
+/**
+ * cors is HTTP middleware that sets an AMP-specific cross origin resource
+ * sharing HTTP header on our response.
+ *
+ * @param {Object} req - The HTTP request object.
+ * @param {Object} res - The HTTP resonse object.
+ * @param {Function} next - The next HTTP handler to execute.
+ */
+function cors(req, res, next) {
+  const host = process.env.NODE_ENV === 'production'
+    ? `https://${req.hostname}`
+    : `http://${req.hostname}:5000`
+
+  res.header('amp-access-control-allow-source-origin', host);
+
+  next();
+}
+
+/**
+ * Cast a value to an array.
+ *
+ * The value is coerced to an array, in the case that the provided value is
+ * null, undefined or an empty string, an empty array is returned.
+ * @param  {any} val - any value.
+ * @returns {array} an array.
+ */
+function castArray(val) {
```

Comment:
toArray?

Reply:
Are you asking to re-name this function?

Reply:
yes
on second thought `convertToArray` to distinguish from the original toArray()